test/unittests/state_transition_create_test.cpp

```diff
@@ -49,3 +49,115 @@ TEST_F(state_transition, create2_max_nonce)
 
     expect.post[*tx.to].nonce = pre.get(*tx.to).nonce;  // Nonce is unchanged.
 }
+
+TEST_F(state_transition, code_deployment_out_of_gas_tw)
+{
+    rev = EVMC_TANGERINE_WHISTLE;  // 63/64 gas rule enabled
+    block.base_fee = 0;
+    const auto initcode = ret(0, 5000);  // create contract will a lot of zeros, deploy cost 1M
+
+    tx.to = To;
+    tx.gas_limit = 1000000;
+    pre.insert(To, {.code = mstore(0, push(initcode)) +
+                            sstore(0, create().input(32 - initcode.size(), initcode.size()))});
+
+    expect.post[To].storage[0x00_bytes32] = 0x00_bytes32;
+}
+
+TEST_F(state_transition, code_deployment_out_of_gas_f)
+{
+    rev = EVMC_FRONTIER;
+    block.base_fee = 0;
+    const auto initcode = ret(0, 1000);  // create contract will a lot of zeros
+
+    tx.to = To;
+    tx.gas_limit = 100000;
+    pre.insert(To, {.code = mstore(0, push(initcode)) +
+                            sstore(0, create().input(32 - initcode.size(), initcode.size()))});
+
+    const auto created = compute_create_address(To, pre.get(To).nonce);
+    expect.post[created].code = bytes{};  // code deployment failure creates empty account
+    expect.post[created].nonce = 0;
+    bytes32 ccc{};
+    std::copy(std::begin(created.bytes), std::end(created.bytes), &ccc.bytes[12]);
```

Comment:
I made a helper for this in this commit, maybe makes sense to extract https://github.com/ethereum/evmone/pull/553/commits/1681575c9bdbb78d348b66722a63e1bd0a4431ee
(in utils.hpp)

Reply:
Extracted. If approved, I will commit the helper separately to have the bug-fix PR minimal.